mlrun/launcher/base.py

```diff
@@ -248,6 +248,12 @@ def _enrich_run(
         # for other runtimes we need to convert the handler to a string
         if run.spec.handler and runtime.kind not in ["handler", "dask"]:
             run.spec.handler = run.spec.handler_name
+            # The name of MLRun's wrapper is 'handler', which is why the handler function name cannot be 'handler'
+            # it would override MLRun's wrapper
+            if runtime.spec.function_kind == "mlrun" and run.spec.handler == "handler":
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    "Handler function name cannot be 'handler'"
```

Comment:
```suggestion
                    "Handler function name 'handler' is preserved. Use different name instead."
```

also, looks like code dup, could it be placed on a single place where both base.py and run.py use and validate their runtime spec?

Reply:
Reserved* (not preserved)

Reply:
also this error will be fixed (if needed) by - https://iguazio.atlassian.net/browse/ML-7278